### ISSUE TITLE
Add number of processed IaC Sources to telemetry

### ISSUE THREAD

**Issue by sjourdan** [snyk/driftctl#1317]
**Description**

TLDR: let's add the number of processed states to the telemetry

Background: 

- when working on improving user journeys, a source is to look at the telemetry
- telemetry sometimes reports difficult to understand behavior, as simultaneous runs with hugely varying number of resources (total/managed)
- we suspect it could be a signal of users using different combinations of states, or gradually onboarding
- a way to know more with full respect for the users' privacy is to add a new telemetry field

**Example**

```json 
{
    "version": "0.7.1",
    "os": "linux",
    "arch": "amd64",
    "iac_source_count": 1
    "total_resources": 100,
    "total_managed": 10,
    "duration": 300,
    "provider_name": "aws"
}
```

**Comment by sundowndev** [comment]
I get the use case and agree with it, I just have a remark on the wording here. I think we should avoid using Terraform wording as much as possible, so the logic/wording stays agnostic and fit with any new IaC supplier we might add in the future (Pulumi, CFN...). Why not name this field with something like `iac_source_count` instead ?

**Comment by sjourdan** [comment]
great idea! let's do this. I updated the main example.

**Comment by sundowndev** [comment]
Released in [v0.20.0](https://github.com/snyk/driftctl/releases/tag/v0.20.0).
